python/cudf/cudf/core/column/string.py

```diff
@@ -2821,6 +2827,12 @@ def endswith(self, pat, **kwargs):
             result_col = column.column_empty(
                 len(self._column), dtype="bool", masked=True
             )
+        elif is_list_like(pat) or isinstance(
+            pat, (cudf.Series, cudf.Index, pd.Series, pd.Index)
+        ):
```

Comment:
May want to handle a `Column` here as well. Should probably put this into a utility function.